### PR TITLE
adds reenqueue by name processor

### PR DESCRIPTION
**Pull request by azharkhan**
We need a way to process large number of deadletter events by filtering specific events. 

This processor just takes an event name as a property and enqueues matching events.

### CODE REVIEW

File: test/deadLetterProcessors.js [hunk 1]
```diff
@@ -56,6 +56,30 @@ describe('Deadletter processors', function() {
     });
   });
 
+  // eslint-disable-next-line
```
Comment (earobinson) [inline test/deadLetterProcessors.js]: why is this needed?

File: lib/deadletterProcessors/reenqueueByName.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const deadletterActions = require('./../deadletterActions');
```
Comment (earobinson) [inline lib/deadletterProcessors/reenqueueByName.js]: minor `reenqueueAllByName`

File: test/deadLetterProcessors.js (hunk 1 above)
Comment (azharkhan) [reply test/deadLetterProcessors.js]: long story short, it's not needed. 

long story long: i thought it was jasmine so I was using fdescribe. little did I know it was mocha (also these tests are FAST)

File: lib/deadletterProcessors/reenqueueByName.js (hunk 2 above)
Comment (azharkhan) [reply lib/deadletterProcessors/reenqueueByName.js]: I thought about it but seemed misleading. so I removed the `All` although I can be argued eitherways. don't feel too strongly about it.

**Comment by azharkhan** [comment]
@earobinson - updated with less `eslint`

File: lib/deadletterProcessors/reenqueueByName.js (hunk 2 above)
Comment (earobinson) [reply lib/deadletterProcessors/reenqueueByName.js]: nope, its a minor comment

File: lib/deadletterProcessors/reenqueueByName.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+const Q = require('q');
+
+module.exports = function(event, message) {
+  const content = JSON.parse(message.content);
+
+  if (event === content.event) {
+    return Q.when(deadletterActions.REENQUEUE_MESSAGE);
+  } else {
+    return Q.reject(new Error(`Required ${event} does not match ${message.content.event}`));
```
Comment (iain8) [inline lib/deadletterProcessors/reenqueueByName.js]: not sure we need `Q` for these, should be able to use native promises

File: test/deadLetterProcessors.js [hunk 4]
```diff
@@ -56,6 +56,29 @@ describe('Deadletter processors', function() {
     });
   });
 
+  describe('Re-Enqueue by Name Processor', function() {
+
+    it('should return with a promise containing requeue all messages action when the event name matches', function() {
+      const testScope = this;
```
Comment (iain8) [inline test/deadLetterProcessors.js]: i don't feel that aliasing the scope is necessary for using it in these tests

File: lib/deadletterProcessors/reenqueueByName.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const deadletterActions = require('./../deadletterActions');
+const Q = require('q');
+
+module.exports = function(event, message) {
+  const content = JSON.parse(message.content);
```
Comment (iain8) [inline lib/deadletterProcessors/reenqueueByName.js]: `parse()` will throw a cheeky error if the message content isn't valid JSON, will this be handled gracefully by the library?

Comment (earobinson) [reply lib/deadletterProcessors/reenqueueByName.js]: in this context iirc the message is guaranteed to be valid json, also this would just cause our commandline app to crash

File: lib/deadletterProcessors/reenqueueByName.js (hunk 3 above)
Comment (azharkhan) [reply lib/deadletterProcessors/reenqueueByName.js]: while I agree with that and I'd use native as much as possible, but I'm not sure (maybe I'm unaware) of a `Q.when()` implementation in the Promise prototype.

Comment (iain8) [reply lib/deadletterProcessors/reenqueueByName.js]: in this case the parameter is just a string so `Promise.resolve()` could work? i find the `when` method mysterious indeed

Comment (azharkhan) [reply lib/deadletterProcessors/reenqueueByName.js]: @earobinson @iain8 - I've created an issue for this https://github.com/hubba/magic-service-bus/issues/10 in order to properly investigate this.

since the `Q` implementation is in step with the rest of the library, I think investigating it is definitely beneficial to reduce unnecessary dependancies.
